src/auto-complete.jsx

```diff
@@ -0,0 +1,295 @@
+const React = require('react');
+const ReactTransitionGroup = require('react-addons-transition-group');
+const StylePropable = require('./mixins/style-propable');
+const ClickAwayable = require('./mixins/click-awayable');
+const KeyCode = require('./utils/key-code');
+const TextField = require('./text-field');
+const Menu = require('./menus/menu');
+const MenuItem = require('./menus/menu-item');
+const MenuDivider = require('./menus/menu-divider');
+
+const AutoComplete = React.createClass({
+
+  mixins: [
+    StylePropable,
+    ClickAwayable,
+  ],
+
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  propTypes: {
+    errorText: React.PropTypes.string,
+    floatingLabelText: React.PropTypes.string,
+    errorStyle: React.PropTypes.object,
+    hintText: React.PropTypes.string,
+    searchText: React.PropTypes.string,
+    dataSource: React.PropTypes.array,
+    updateWhenFocused: React.PropTypes.bool,
+    auto: React.PropTypes.bool,
+    id: React.PropTypes.string,
+    autoWidth: React.PropTypes.bool,
+    menuStyle: React.PropTypes.object,
+    listStyle: React.PropTypes.object,
+    menuProps: React.PropTypes.object,
+    menuCloseDelay: React.PropTypes.number,
+    onUpdateRequests: React.PropTypes.func,
+    onNewSearchRequest: React.PropTypes.func,
+    disableFocusRipple: React.PropTypes.bool,
+  },
+
+  getDefaultProps() {
+    return {
+      fullWidth: false,
+      open: false,
+      auto: false,
+      searchText: '',
+      menuCloseDelay: 100,
+      disableFocusRipple: true,
+      updateWhenFocused: false,
+      onUpdateRequests: () => {},
```

Comment:
```onUpdateRequests ``` is used to request remote update from server,  dataSource is used for static data list. they are designed for different purpose. I think the first case will be more straightforward in many situations.